engine/crates/engine/registry-upgrade/src/lib.rs

```diff
@@ -69,6 +69,8 @@ pub fn convert_v1_to_v2(v1: registry_v1::Registry) -> registry_v2::Registry {
 
     // Build a map of type name -> the ID it'll have when we insert it.
     let preallocated_ids = writer.preallocate_type_ids(types.len()).collect::<Vec<_>>();
+    assert_eq!(preallocated_ids.len(), types.len());
```

Comment:
weird assert to have in production code, any reason this can fail at runtime?

Reply:
I put this in to debug a fuckup I'd made, and I'm leaving it in incase that fuckup re-appears.

I'd only expect this to fail if someone messes with the `ID`/`IdRange`/`Iter` implementations in such a way that they're wrong.

Reply:
Sounds like you should have unit tests for whatever you rely on IMHO, but 🤷